### PR TITLE
style: add z-index to modals

### PR DESCRIPTION
**Pull request by danidewitt**
## What did we change?
- style: add z-index to EzModal
- docs: add theming documentation about z-indices and layering
- docs: remove all localhost:8000 references from docs

## Why are we doing this?
Modals should appear above app bars in downstream applications using z-indices.

## Screenshot(s) / Gif(s):
Screen Shot 2022-08-31 at 1 23 42 PM
Screen Shot 2022-08-31 at 1 24 16 PM
Screen Shot 2022-08-31 at 1 25 19 PM

## Checklist

- [ ] Provide test coverage for the changes made
- [x] Create a changeset (`yarn changeset`) with a summary of the changes

[Contributing guidelines](https://ezcater.github.io/recipe/guides/contributing/)

### CODE REVIEW

**Review by pgrippi — APPROVED** [review]

**Review by noranda — APPROVED** [review]
Looks good!

**Review by pgrippi — APPROVED** [review]

**Comment by danidewitt** [comment]
@noranda updated `EzDateInput` -[ this is a good code block](https://ezcater.github.io/recipe/playroom/?code=N4Igxg9gJgpiBcJgApkEoAEBeAfB4AOgHYYaREDOALhgNoCWFA8gA4xEA0GFMVAkszZEAutgwAlGAEMwVAHQBXHgGUqUqjGQAzKQBseaANzFS5anUbKFAIwC29KlXpEA5lx78KVuw6evRWBLSsooqahrIRAq6ukYmZBCUNORqzjAATpJaYpIy8mDp0hpZ6MZE8WY0jADqMDAA1lBSAJ5iTRrYeIQkpAlJGE2tgUQwAO4YACLqmu0waHIuvFPNpfGkhVQK6SSDGACEWIEADBgAZKcDLfuHGABsZaQAvmXxG1skyGsYADw4X6TfACiAC8AEIKRyJDCJADCunoYHqWBQmFw3F4AlY7GQVHSCjmjxwWKI3wA9CDwZCiH8er0fiCALLQPT-XqWGz2RzOFzI9k%2BLmuR6s0iMYm8wTsIW0unQojeTnI9CdfDC3oeATy3zcnF4uYPGV0jwAFXothgEAhqFRXVVhoxXg5Wtc2j0Bn1BrtnmJLv0ettGEeXAATEdQ3FpXTHlKPdxHVQADJSawwXRYAggTVUdOqxITRj2CgURXW9FeoQ%2BgzRj1QfOMCiJ5Op9N5igFijZiO9AAW0lg6SNMAAHlQ0yAABK9jIYFwQGAUDA9wodg00mMAmsANwwhS0yJSUjSmRgWkJtv9dKBwIAYvQU1BzzLdEmU6OmOk%2B5cNMu1watPRdBo6RTBo4q1A0gxVj%2BdJUM0bCjrM35Qb0G56Pio5HAAjKSIbYUcIaIUhGBSBCEBXhAYBKA%2BpCkqudL%2BmSm60Re5LAkyTS6ExPw0fE4aPGg6CPCAHAgKM9BQFQXYUAgtAAMwhsIjxAA) to use to test it, if you wanted to do that locally, otherwise chromatic should pass now.

File: packages/recipe/src/components/EzField/EzDateInput.tsx [hunk 1]
```diff
@@ -17,6 +17,10 @@ const layout = theme.css({
   input: {paddingLeft: '2.5em'},
 });
 
+const popper = theme.css({
+  zIndex: '$date-input-z', // this is hard-coded in theme.config.ts for now, but we should really pull it in from mui when we convert modals to mui
```
Comment (noranda) [inline packages/recipe/src/components/EzField/EzDateInput.tsx:21]: Should the comment say "... when we convert date inputs to mui"?

File: packages/recipe/src/components/theme.config.ts [hunk 2]
```diff
@@ -840,6 +843,7 @@ const stitches = createStitches({
       // Icons
       'icon-inset-z1': 1,
       'modal-z': 1300, // this is hard-coded in theme.config.ts for now, but we should really pull it in from mui when we convert modals to mui
+      'date-input-z': 1350, // this is hard-coded in theme.config.ts for now, but we should really pull it in from mui when we convert modals to mui
```
Comment (noranda) [inline packages/recipe/src/components/theme.config.ts:846]: Should these be under a different section than "Icons"?

**Review by noranda — APPROVED** [review]
The issue is fixed. Just a few minor comments and it needs a rebase, but otherwise looks good!

**Review by pgrippi — DISMISSED** [review]

**Review by pgrippi — APPROVED** [review]

**Review by noranda — APPROVED** [review]

**Pull request merged**